src/Microsoft.TestPlatform.Common/Utilities/RunSettingsProviderExtensions.cs

```diff
@@ -62,6 +89,61 @@ public static void UpdateRunSettingsNode(this IRunSettingsProvider runSettingsPr
             runSettingsProvider.UpdateRunSettings(xmlDocument.OuterXml);
         }
 
+        /// <summary>
+        /// Matches with test run parameter node pattern and returns that match.
+        /// </summary>
+        /// <param name="runSettingsProvider"></param>
+        /// <param name="node"></param>
+        /// <returns></returns>
+        public static Match GetTestRunParameterNodeMatch(this IRunSettingsProvider runSettingsProvider, string node)
+        {
+            var attrName = $"(?<{AttributeNameString}>\\w+)";
+            var attrValue = $"(?<{AttributeValueString}>.+)";
+            Regex regex = new Regex($"{Constants.TestRunParametersName}.{ParameterString}\\(name\\s*=\\s*\"{attrName}\"\\s*,\\s*value\\s*=\\s*\"{attrValue}\"\\)");
+            Match match = regex.Match(node);
```

Comment:
this can throw exception, either catch here, or in the caller.

Reply:
There are four argument exceptions possible
1.ArgumentOutOfRangeException : This comes if options is not a valid bitwise combination, but we have not given options in match
2.ArgumentNullException : This commes if input or pattern is null, but input is always not null and pattern is not null.
3.RegexMatchTimeoutException :This commes if timout occured,but as Pattern is fixed and we have not set timeout in match. We won't get this exceptions
4.ArgumentException : This comes if regular expression is not syntactically correct,This too not valid in our case..
so catching an exception is not that reasonable